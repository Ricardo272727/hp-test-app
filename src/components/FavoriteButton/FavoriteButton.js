import React from "react";
import "./FavoriteButton.scss";

export const FavoriteButton = ({ selected, onClick }) => {
  return (
    <button
      className={`favorite-button ${selected ? "selected" : ""}`}
      onClick={onClick}
    ></button>
  );
};
