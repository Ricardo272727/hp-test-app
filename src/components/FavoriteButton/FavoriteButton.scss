.favorite-button{
  width: 1.5rem;
  height: 1.5rem;
  background-image: url(assets/favorite-icon.png);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center center;
  border: none;
  background-color: transparent;

  &:hover{
    cursor: pointer;
    outline: none;
  } 
  &.selected{
    background-image: url(assets/favorite-selected-icon.png);
  }
}
