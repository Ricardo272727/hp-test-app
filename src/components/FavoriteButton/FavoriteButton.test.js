import React from 'react';
import renderer from "react-test-renderer";
import { FavoriteButton } from "./FavoriteButton";

test('toggle selected class for favorite button', () => {
  const fn = jest.fn((e) => {});
  const component = renderer.create(<FavoriteButton onClick={fn} selected />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.selected = false;
  expect(tree).toMatchSnapshot();
});
