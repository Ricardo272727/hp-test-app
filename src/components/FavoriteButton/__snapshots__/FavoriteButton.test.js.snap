// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggle selected class for favorite button 1`] = `
<button
  className="favorite-button selected"
  onClick={[MockFunction]}
/>
`;

exports[`toggle selected class for favorite button 2`] = `
<button
  className="favorite-button selected"
  onClick={[MockFunction]}
/>
`;
